### PR TITLE
Modernise Manage Jenkins screen

### PR DESCRIPTION
**Pull request by janfaracik**
### Proposed changelog entries

* Modernise the "Manage Jenkins" screen

### Screenshot

Old
image

New
image

Hover/focus states have been updated to match buttons, text and padding sizes have been tweaked to make page more readable and coherent. 

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]
Looks fine with Dark Theme too, thanks!

image

**Comment by daniel-beck** [comment]
Design wise, this looks nice.

---

Could you clarify why you decided to remove the scaling based on window width? This now is always 3 columns wide, rather than scaling from 1 to 4, depending on available space on the screen. This was a deliberate decision so that individual entries are not overly squished, no matter the screen size.

Extremes, before (alt text for description):

> Most narrow, 2 columns
> Widest, 4 columns

Extremes, after:

> With sidepanel, 3 columns, narrow
> With sidepanel, 3 columns, wide

---

When the layout moves content below the "side"panel, it previously went to 1 column width, i.e. a basic list. With this PR it's still 3 columns wide. Of course it's a valid question to ask how often this is relevant, but I think recent work even on forms was to make it more mobile/small screen friendlier, so this looks like a step back for that.

> Screenshot 2021-08-28 at 10 56 40

> Screenshot 2021-08-28 at 11 03 24

**Comment by janfaracik** [comment]
> Wouldn't it make sense to increase the font size of the sections? Currently the section titles (h2 tag) are much smaller than the heading and texts of the individual buttons.

It was solely a design decision, somewhat inspired by Material Design or iOS - 

image

image

**Comment by janfaracik** [comment]
> Design wise, this looks nice.
> 
> Could you clarify why you decided to remove the scaling based on window width? This now is always 3 columns wide, rather than scaling from 1 to 4, depending on available space on the screen. This was a deliberate decision so that individual entries are not overly squished, no matter the screen size.
> 
> Extremes, before (alt text for description):
> 
> > Most narrow, 2 columns
> > Widest, 4 columns
> 
> Extremes, after:
> 
> > With sidepanel, 3 columns, narrow
> > With sidepanel, 3 columns, wide
> 
> When the layout moves content below the "side"panel, it previously went to 1 column width, i.e. a basic list. With this PR it's still 3 columns wide. Of course it's a valid question to ask how often this is relevant, but I think recent work even on forms was to make it more mobile/small screen friendlier, so this looks like a step back for that.
> 
> > Screenshot 2021-08-28 at 10 56 40
> 
> > Screenshot 2021-08-28 at 11 03 24

Hm that's behaving a bit differently on my end, I'll restore the original way so that it behaves the correct way 👍

**Comment by daniel-beck** [comment]
FWIW we've generally tried to keep PR diffs minimal to make changes easier to review, cherry-pick, and merge -- even if the overall file format was inconsistent and wrong.

**Comment by oleg-nenashev** [comment]
> FWIW we've generally tried to keep PR diffs minimal to make changes easier to review, cherry-pick, and merge -- even if the overall file format was inconsistent and wrong.

This is not a bugfix, and there are no pending pull requests against this screen. I am fine with keeping refactoring integrated though I agree with @daniel-beck in general

**Review by timja — APPROVED** [review]
Lgtm

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by daniel-beck** [comment]
I'd like an opportunity to test this PR again before it's merged.

**Comment by res0nance** [comment]
Tested on I.E

I see this issue
image

Hovering over system log seems to cause the button to overlap?

**Comment by timja** [comment]
> I'd like an opportunity to test this PR again before it's merged.

Sure, I tested it on my 27” monitor and it was fine

**Comment by daniel-beck** [comment]
Is it deliberate that administrative monitor warnings appear between horizontal bar and "System Configuration" title? Looking at the screenshots I got the impression of horizontal bars and titles as a unit, but they're split up at the first one.

> Screenshot 2021-09-01 at 14 18 31

I'm not rejecting it, it's just something I noticed.

**Comment by janfaracik** [comment]
> Tested on I.E
> 
> I see this issue
> image
> 
> Hovering over system log seems to cause the button to overlap?

It's due to the lack of gap between the buttons sadly - it turns out that IE11 doesn't support `grid-gap` annoyingly so I've added a workaround that adds margins to the buttons for IE11 - should solve that issue for you :)

**Comment by janfaracik** [comment]
> Is it deliberate that administrative monitor warnings appear between horizontal bar and "System Configuration" title? Looking at the screenshots I got the impression of horizontal bars and titles as a unit, but they're split up at the first one.
> 
> > Screenshot 2021-09-01 at 14 18 31
> 
> I'm not rejecting it, it's just something I noticed.

image

Ah it was originally like that, must have broken it by mistake when changing the header. I've restored the OG look :)

**Comment by daniel-beck** [comment]
@janfaracik Thanks for the super quick reaction!

**Comment by daniel-beck** [comment]
> @daniel-beck are you wanting to look again?

Yes.

**Review by StefanSpieker — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
Gentle ping @daniel-beck. There is no time pressure for the next LTS baseline, but it would be awesome to merge it. There are many Web UI PRs coming it, and I'd rather avoid merge conflicts

**Comment by timja** [comment]
reminder @daniel-beck.

**Comment by daniel-beck** [comment]
I tested the current PR and it works well. My feedback has been addressed fully, thanks for those changes. I like in particular the rounded corners and "active" border around entries on this page. The headers look good IMO in isolation. As far as I can tell (not a CSS expert), the implementation is done well too.

If this is the first part of a larger scale planned redesign involving forms etc., it's a very promising start and I look forward to future changes.

If this is all that's planned, I am not sure the modernization is worth the additional inconsistency with other parts of the Jenkins UI. While a few places with vertical dividers exist (e.g. with nested sidepanel items, _below_ section headers on config forms), they're not used like these are, and the small all-caps headers are a completely new element in Jenkins. I realize this screen has always been different from others, and I contributed to making it even more so in the past, but that was mostly elements that didn't exist elsewhere. The parts that have equivalents elsewhere (headers) remained similar.

I don't feel qualified enough in the UI space to either approve or reject this PR as a whole.

**Review by uhafner — COMMENTED** [review]
It looks better than before, so I am fine with merging this PR. 

But I am still unsure about the new design of the section headers. I think the texts are too small and all caps texts are not used in Jenkins up to now. And they are not consistent with the rest of Jenkins UI (they weren't before either). 

E.g. in the configure screens we have:

Bildschirmfoto 2021-09-13 um 23 20 13
  
This looks better IMHO.

**Comment by fqueiruga** [comment]
I think this looks great, cheers! Heading consistency is a problem, but I can see this page as a one-off screen regarding some design elements.

File: war/src/main/less/base/layout-commons.less
Comment (fqueiruga) [inline war/src/main/less/base/layout-commons.less:79]: might as well use `2rem` here

Comment (fqueiruga) [reply war/src/main/less/base/layout-commons.less:79]: Also, I'm a bit -1 on this change. This will affect everything, including page footers, etc.

File: war/src/main/less/modules/app-bar.less
Comment (fqueiruga) [inline war/src/main/less/modules/app-bar.less:1]: What does app-bar mean? If it's for the configure screen, it should have a component-specific name. We already have CSS for the manage jenkins page.

File: war/src/main/less/modules/app-bar.less
Comment (fqueiruga) [inline war/src/main/less/modules/app-bar.less:30]: We should style heading sizes using the `.h1`, `.h2` classes, etc. Also, font-size should be in rems in any case.

File: war/src/main/less/modules/manage-jenkins.less
Comment (fqueiruga) [inline war/src/main/less/modules/manage-jenkins.less:63]: What does `.jenkins-settings` mean? Sorry for being cheeky, but if it's just for the manage page I'd just name it like it is, not a generic jenkins settings component.

Comment (fqueiruga) [reply war/src/main/less/modules/manage-jenkins.less:63]: I would guess it would apply to the general settings page, or plugin manager for example.

File: war/src/main/less/modules/manage-jenkins.less
Comment (fqueiruga) [inline war/src/main/less/modules/manage-jenkins.less:64]: I'm always a proponent of using unexploded classnames. I think SCSS's nesting hurts maiteinability. Would you mind using fully-qualified selectors like `jenkins-settings__section-title` or `.jenkins-settings__section?

File: war/src/main/less/modules/manage-jenkins.less
Comment (fqueiruga) [inline war/src/main/less/modules/manage-jenkins.less:82]: This is sort of what I meant, tracking the `.jenkins-settings__section .mange-page__row` to this file would be very quite hard doing file search.

File: war/src/main/less/modules/manage-jenkins.less
Comment (fqueiruga) [inline war/src/main/less/modules/manage-jenkins.less:77]: I'm fine with leaving pseudoselectors nested.

File: war/src/main/less/modules/manage-jenkins.less
Comment (fqueiruga) [inline war/src/main/less/modules/manage-jenkins.less:132]: Can't remember now, but doesn't this also have variables for button link color and border?

File: war/src/main/less/modules/manage-jenkins.less
Comment (fqueiruga) [inline war/src/main/less/modules/manage-jenkins.less:147]: Is this rule nested under `.jenkins-settings`?

**Review by fqueiruga — CHANGES_REQUESTED** [review]
I have a few comments regarding CSS style and naming for the classes. `.jenkins-settings` feels a bit too generic to me.

The only change I can consider a must is revisit the padding of the `#main-panel`. It can have side effects across quite a few plugins and screens.

**Comment by janfaracik** [comment]
> If this is the first part of a larger scale planned redesign involving forms etc., it's a very promising start and I look forward to future changes.
> 
> If this is all that's planned, I am not sure the modernization is worth the additional inconsistency with other parts of the Jenkins UI. While a few places with vertical dividers exist (e.g. with nested sidepanel items, _below_ section headers on config forms), they're not used like these are, and the small all-caps headers are a completely new element in Jenkins. I realize this screen has always been different from others, and I contributed to making it even more so in the past, but that was mostly elements that didn't exist elsewhere. The parts that have equivalents elsewhere (headers) remained similar.

I've got some branches on my fork of Jenkins at the moment looking at improving the interface across the service - updating Jenkins' forms definitely plays a large part in that. Current WIP for that below:

Jenkins Forms

> It looks better than before, so I am fine with merging this PR.
> 
> But I am still unsure about the new design of the section headers. I think the texts are too small and all caps texts are not used in Jenkins up to now. And they are not consistent with the rest of Jenkins UI (they weren't before either).

As for the headers, I'm removed the all caps text and increased the size so that it's consistent with the rest of Jenkins.

It looks like so -

image

Hope that addresses your comments :)

File: war/src/main/less/modules/app-bar.less
Comment (janfaracik) [reply war/src/main/less/modules/app-bar.less:1]: The app bar represents the heading for the page, it contains the title (in this case Manage Jenkins) with the option of placing components on the far right of it. I wanted to get it merged in with this MR as I plan on making extensive use of it in future design related MRs.

File: war/src/main/less/base/layout-commons.less
Comment (janfaracik) [reply war/src/main/less/base/layout-commons.less:79]: I've changed it to 2rem. The padding will affect every page, but I think it's worth it as it improves the spacing of Jenkins' UI and gives it more balance.

File: war/src/main/less/modules/manage-jenkins.less
Comment (janfaracik) [reply war/src/main/less/modules/manage-jenkins.less:63]: I named it `jenkins-settings` over `jenkins-manage/jenkins-manage-jenkins/manage-jenkins` or something along those lines as it's a touch shorter and more readable. I do recognise that it might not be instantly clear however, I think moving the file into a `pages` package over the current `modules` package could help devs understand that its styles for a specific page rather than a component?

File: war/src/main/less/modules/manage-jenkins.less
Comment (janfaracik) [reply war/src/main/less/modules/manage-jenkins.less]: Done :)

File: war/src/main/less/modules/manage-jenkins.less
Comment (janfaracik) [reply war/src/main/less/modules/manage-jenkins.less]: As mentioned above, that styling as now been moved out into its own file under `.jenkins-section`

File: war/src/main/less/modules/manage-jenkins.less
Comment (fqueiruga) [reply war/src/main/less/modules/manage-jenkins.less]: Thanks!

File: war/src/main/less/modules/app-bar.less
Comment (fqueiruga) [inline war/src/main/less/modules/app-bar.less:7]: nice to see CSS vars <3

File: war/src/main/less/modules/app-bar.less
Comment (fqueiruga) [inline war/src/main/less/modules/app-bar.less:6]: I would suggest this could be changed to `2rem`. All `30px` instances really.

**Comment by timja** [comment]
> hudson.model.ManagementLinkTest.links failed

https://github.com/jenkinsci/jenkins/pull/5693/checks?check_run_id=3646260784
I assume related ^^

**Review by fqueiruga — APPROVED** [review]
I'm happy to hear that there are intentions of working on forms. I think it's needed. I think that would be great, especially if all stuff like input border, padding, height, etc is configurable.

Regarding the reusability of the headings... As @daniel-beck commented on https://github.com/jenkinsci/jenkins/pull/5693#issuecomment-918517681, this screen has always be a one-off. I see nothing wrong in having a one-off header just for this page. It can also be tricky to do.

Thanks for the PR! Approving the code based on how it looks, haven't tried it. Some specific feedback to note is that dimensions above 4-6px should probably be in `rem`s IMO. I love the use of existing CSS vars.

Cheers!

**Comment by timja** [comment]
Not sure if I'm a fan of the supporting text weight change:

(Looks like the `font-weight` changed from 400 to 600)

Before:

image

After:

image

Imo it especially doesn't look so good when there's a lot of text:

image

any thoughts on that ^?

Apart from that all looks good to me

**Comment by fqueiruga** [comment]
I agree @timja, it's a subtitle after all.

File: war/src/main/less/modules/section.less
Comment (timja) [inline war/src/main/less/modules/section.less:104]: this should fix the issue I'm seeing with the text being too bold
```suggestion
    font-weight: 400;
```

**Comment by janfaracik** [comment]
> > hudson.model.ManagementLinkTest.links failed
> 
> https://github.com/jenkinsci/jenkins/pull/5693/checks?check_run_id=3646260784
> I assume related ^^

Strange, I ran the test in IntelliJ and all was fine :< I've updated the selector anyhow to use the new class name,  hopefully that'll fix it

File: war/src/main/less/modules/app-bar.less
Comment (janfaracik) [reply war/src/main/less/modules/app-bar.less]: Done :)

File: war/src/main/less/modules/section.less
Comment (janfaracik) [reply war/src/main/less/modules/section.less]: Done :)

**Comment by timja** [comment]
> > > hudson.model.ManagementLinkTest.links failed
> > 
> > 
> > [#5693 (checks)](https://github.com/jenkinsci/jenkins/pull/5693/checks?check_run_id=3646260784)
> > I assume related ^^
> 
> Strange, I ran the test in IntelliJ and all was fine :< I've updated the selector anyhow to use the new class name, hopefully that'll fix it

probably cached jelly files in some way, sometimes IntelliJ goes quite crazy with core tests.
Clean build or invalidating caches might fix it.

Sometimes running tests in intellij works fine and sometimes you get non-sensical errors 😢 .

**Review by timja — APPROVED** [review]
looks better now

image

**Comment by timja** [comment]
@daniel-beck did you want to take a last look at this?

**Review by daniel-beck — APPROVED** [review]
Looks good. Thanks!

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by dheerajodha** [comment]
Proposed changelog entries

- Modernise Manage Jenkins screen

can be changed to:

- Modernise the <code>Manage Jenkins</code> screen.

to put emphasis on the screen which was changed.

**Comment by Bananeweizen** [comment]
@janfaracik Please don't blindly copy design from your phone. The slider element in your WIP screenshot "Publish results with TM4J") is a rather bad choice for checkbox input elements in an application like Jenkins that is mainly used on operating systems _without_ touch. The metaphor of sliding left and right doesn't fit when most people use a mouse click for it.

**Pull request merged**
